packages/components/src/Button/GenericButton/GenericButton.tsx

```diff
@@ -258,32 +262,38 @@ const renderButton = (
     formNoValidate: props.formNoValidate,
     className: buttonClass(props),
     "aria-label": generateAriaLabel(props),
-    "aria-disabled": props.disabled || props.working ? true : undefined,
-    tabIndex: props.disableTabFocusAndIUnderstandTheAccessibilityImplications
-      ? -1
-      : undefined,
     ...getCustomProps(props),
   }
 
-  // we're using useFocusable instead of useButton because at this stage we want to hook only to focusable.
-  // Not standardize button behavior as we're currently relying on some weird native behaviours (like onClick firing on enter key press) see https://react-spectrum.adobe.com/blog/building-a-button-part-1.html
-  // @ts-ignore
-  const { focusableProps } = useFocusable(passedInProps, ref)
+  const [contextProps, contextRef] = useContextProps(
+    passedInProps,
+    ref,
+    ButtonContext
+  )
+
+  const { buttonProps } = useButton(contextProps, contextRef)
 
   return (
     // eslint-disable-next-line react/button-has-type
     <button
-      {...passedInProps}
-      {...focusableProps}
+      {...contextProps}
+      {...buttonProps}
+      className={buttonProps.className as string}
+      slot={buttonProps.slot as string | undefined}
+      style={buttonProps.style as CSSProperties}
+      tabIndex={
+        props.disableTabFocusAndIUnderstandTheAccessibilityImplications
+          ? -1
+          : undefined
+      }
+      onMouseDown={!disableActions ? props.onMouseDown : undefined}
```

Comment:
shouldn't this come from buttonProps?
```suggestion
      onMouseDown={!disableActions ? buttonProps.onMouseDown : undefined}
```